property/index.js

```diff
@@ -112,53 +121,158 @@ module.exports = yeoman.Base.extend({
-        if (['true', '1', 't'].indexOf(this.defaultValue) !== -1 ){
-          def.default = true;
-        } else {
-          def.default = false;
-        }
-      } else if (this.defaultValue === 'uuid' || this.defaultValue === 'guid'){
-         def.defaultFn = this.defaultValue;
-      } else if ((this.type === 'date' || this.type === 'datetime') &&
-                  this.defaultValue.toLowerCase() === 'now'){
-         def.defaultFn = 'now';
-      } else {
-         def.default = this.defaultValue;
-      }
-    }
-
-    this.modelDefinition.properties.create(def, function(err) {
+    this.modelDefinition.properties.create(this.propDefinition, function(err) {
       helpers.reportValidationError(err, this.log);
       return done(err);
     }.bind(this));
   },
-
   saveProject: actions.saveProject
 });
+
+function coerceDefaultValue(propDef, value) {
+  var itemType;
+  if (Array.isArray(propDef.type)) {
+    itemType = propDef.type[0];
+    propDef.type = 'array';
+  }
+
+  switch (propDef.type) {
+    case 'string':
+      if (value === 'uuid' || value === 'guid'){
+        propDef.defaultFn = value;
+      } else {
+        propDef.default = value;
+      }
+      break;
+    case 'number':
+      propDef.default = castToNumber(value);
+      break;
+    case 'boolean':
+      propDef.default = castToBoolean(value);
+      break;
+    case 'date':
+      if (value.toLowerCase() === 'now'){
+        propDef.defaultFn = 'now';
+      } else {
+        propDef.default = castToDate(value);
+      }
+      break;
+    case 'array':
+      propDef.type = [itemType];
+      if (itemType === 'string') {
+        propDef.default = value.replace(/[\s,]+/g, ',').split(',');
+      } else if (itemType === 'number') {
+        propDef.default = value.replace(/[\s,]+/g, ',').split(',')
+          .map(function(item) {
+            return castToNumber(item);
+          });
+      } else if (itemType === 'boolean') {
+        propDef.default = value.replace(/[\s,]+/g, ',').split(',')
+          .map(function(item) {
+            return castToBoolean(item);
+          });
+      } else if (itemType === 'date') {
+        propDef.default = value.replace(/[\s,]+/g, ',').split(',')
+          .map(function(item) {
+            return castToDate(item);
+          });
+      } else {
+        propDef.default = value;
+      }
+      break;
+    case 'geopoint':
+      if (value.indexOf('lat') !== -1 && value.indexOf('lng') !== -1) {
+        propDef.default = JSON.parse(value);
+      } else {
+        var geo = value.replace(/[\s,]+/g, ',').split(',');
+        propDef.default = {};
+        propDef.default.lat = castToNumber(geo[0]);
+        propDef.default.lng = castToNumber(geo[1]);
+      }
+      break;
+    case 'object':
+      propDef.default = JSON.parse(value);
+      break;
+    default:
+      propDef.default = value;
+  }
+}
+
+function castToDate(value) {
+  var dateValue;
+  var isNumber = /^[0-9]+$/.test(value);
```

Comment:
Note that this does not support scientific notation, e.g. `1.2e10`, but I think that's not a big deal here.